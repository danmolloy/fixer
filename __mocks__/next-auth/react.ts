import { mockSession } from '../session';

module.exports = {
  __esModule: true,
  ...jest.requireActual('next-auth/react'),
  auth: jest.fn(() => {
    return mockSession;
  }),
  signIn: jest.fn(),
  signOut: jest.fn(),
  useSession: jest.fn(),
};
