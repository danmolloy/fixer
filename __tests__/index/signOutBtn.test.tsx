import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import SignOutBtn from "../../components/index/signOutBtn";
import React from "react";

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual('next-auth/react');
  const mockSession = {}
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return {data: mockSession, status: 'authenticated'}  // return type is [] in v3 but changed to {} in v4
    }),
  };
});


describe("SignOutBtn component", () => {
  beforeEach(() => {
    render(<SignOutBtn />)
  })
  it("Renders", () => {
    const signOutBtn = screen.getByTestId("signout-btn")
    expect(signOutBtn).toBeInTheDocument()
  })
})