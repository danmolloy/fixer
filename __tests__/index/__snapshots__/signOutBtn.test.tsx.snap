// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOutBtn component Matches snapshot 1`] = `
<div
  class="flex flex-col justify-center h-full"
  data-testid="signout-btn"
>
  <button
    class="sign-in-btn justify-center"
  >
    Sign out
  </button>
</div>
`;
